File: nixos/examples/typescript-devshell/hello.ts
// Tiny typed program proving the flake's Node + TypeScript are on PATH.
// Compile and run:  tsc hello.ts && node hello.js
// Both `tsc` and `node` come from the flake's dev shell.

function greet(name: string): string {
  return `Hello, ${name}! The TypeScript dev shell works.`;
}

console.log(greet("world"));

File: nixos/examples/python-devshell/hello.py
#!/usr/bin/env python3
"""Tiny script proving the flake's Python + packages are on PATH.

`requests` is imported (not called) so this runs offline — if the import
succeeds, the dev shell loaded the flake's Python environment correctly.
"""

import sys

import requests  # provided by the flake, not the system Python


def main() -> None:
    print(f"Hello from Python {sys.version.split()[0]}")
    print(f"requests {requests.__version__} is importable — the dev shell works.")


if __name__ == "__main__":
    main()
